src/imap.rs

```diff
@@ -772,6 +776,8 @@ impl Imap {
             );
         }
 
+        chat::mark_old_messages_as_noticed(&ctx, received_msgs).await?;
```

Comment:
I thought about doing this asynchronously (async_std::task::spawn()). However, on iOS, this might go wrong, though it is probably very unlikely:
- First, some background: After fetching messages (and executing jobs) is finished, we go back to `DetailedConnectivity::Connected` and `all_work_done()` returns `true` again. Then the app signals to the system that it can go back to sleep again. The system then terminates the process.
- Now, it might happen in theory that at this point `mark_old_messages_as_noticed()` isn't done yet and the system terminates the process.